hdk/common/verif/scripts/Makefile

```diff
@@ -20,12 +20,23 @@ DEPS_FILE := $(HDK_SHELL_DESIGN_DIR)/ip/ddr4_core/ddr4_core.xci
 DEPS_FILE += $(HDK_COMMON_DIR)/verif/scripts/init.tcl
 DEPS_FILE += $(HDK_COMMON_DIR)/verif/scripts/init.sh
 
+# Replace upstream aws-fpga's use of git clean with an explicit rm -rf
+# Note, it is likely only tmp/ that has to be removed, but all the files
+# that would generally be removed are included here.
+#   git clean -fXdq $(shell dirname $@)
+
+CLEAN_TARGETS += $(HDK_COMMON_DIR)/verif/scripts/ddr4_core_ip_report.txt
+CLEAN_TARGETS += $(HDK_COMMON_DIR)/verif/scripts/tmp/
+# These are inoffensive, but i've put them here for consistency
+CLEAN_TARGETS += $(wildcard $(HDK_COMMON_DIR)/verif/scripts/vivado*.jou)
+CLEAN_TARGETS += $(wildcard $(HDK_COMMON_DIR)/verif/scripts/vivado*.log)
+
 all: $(DONE_FILE)
 
 $(DONE_FILE): $(DEPS_FILE)
 	@ echo "INFO: Building in $(shell dirname $@)"
 	@ echo "INFO: This could take 5-10 minutes, please be patient!"
-	@ git clean -fXdq $(shell dirname $@)
+	@ rm -rf $(CLEAN_TARGETS)
```

Comment:
I agree in theory that we should do this because needing to haul along the git repo to build this is insane.

I worry that maintaining the list of CLEAN_TARGETS as you've implemented it will be onerous.  Can we tweak the implementation so that there is an automated way to generate the cleanup command that we can run after merging?

I'm thinking something like adding a script called `firesim_clean.sh` in this directory and putting a target here that looks like:

```make
.PHONY: $(HDK_COMMON_DIR)/verif/scripts/firesim_clean #yeah, it's autogenerated and checked in, so sue me
$(HDK_COMMON_DIR)/verif/scripts/firesim_clean: $(DONE_FILE)
        git clean -fXdn $(shell dirname $@) |& awk '{<introspect the crap git thinks should be deleted>}' > $@

```
And then have this line `cat firesim_clean | xargs rm -rf` instead of having the list of stuff in the makefile?

NOTE: you can substitute a python script or whatever you would like instead of `awk`.

You could even be more explicit and create a `gen_firesim_clean.py` script and put a README.md in this directory explaining that you need to run the script to fixup the cleanup after merging anything from upstream aws-fpga.  

I suppose if you have a separate script doing the gen, you could have it update the makefile directly, I just wouldn't want to start writing a self-modifying makefile.  That sounds a bit too much like it would need a lambda calculus to get it done right...

Alternately, we could refactor this to run vivado in a 'build' or `run_init` subdirectory that we know can always be blown away by clean and upstream that to aws-fpga.

Perhaps I'm 🚲 shedding there though.

Reply:
Maybe the right solution here is to take this admittedly unsound implementation, and try to upstream it as is, and then AWS can decide how to want to maintain it. 

This code hasn't changed in years and i don't think it's worth spending time to do it "right" considering what the costs of getting it wrong are.

Reply:
I like the sound of that idea.  They are the ones who need to do this better, not us.